libpod/oci_conmon_linux.go

```diff
@@ -669,6 +669,31 @@ func (r *ConmonOCIRuntime) CheckpointContainer(ctr *Container, options Container
 	return utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, nil, r.path, args...)
 }
 
+func (r *ConmonOCIRuntime) CheckConmonRunning(ctr *Container) (bool, error) {
+	if ctr.state.ConmonPID == 0 {
+		// If the container is running or paused, assume Conmon is
+		// running. We didn't record Conmon PID on some old versions, so
+		// that is likely what's going on...
+		// Unusual enough that we should print a warning message though.
+		if ctr.ensureState(define.ContainerStateRunning, define.ContainerStatePaused) {
+			logrus.Warnf("Conmon PID is not set, but container is running!")
+			return true, nil
+		}
+		// Container's not running, so conmon PID being unset is
+		// expected. Conmon is not running.
+		return false, nil
+	}
+
+	// We have a conmon PID. Ping it with signal 0.
+	if err := unix.Kill(ctr.state.ConmonPID, 0); err != nil {
```

Comment:
I am not sure if you're worried about this, but CRI-O has had trouble killing pids directly because of pid wrap. It *could* be worth taking down conmon's start time, and then comparing the ps entry with the start time to make sure it's the same process.
seems a bit like overkill, but I thought I'd mention.

Reply:
I think we wait for this to happen, not likely unless you are running something like Kubernetes where you are constantly creating millions of processes and simultaneously conmon crashes.

Reply:
Looking forward to pid_fs being widely available.